Obsolete C libraries may not provide isnan/isfinite macros

**Checklist (Please check before submitting)**

* [x] I reviewed the [Contributing Guide](https://github.com/nasa/LC/blob/main/CONTRIBUTING.md).
* [x] I performed a cursory search to see if the bug report is relevant, not redundant, nor in conflict with other tickets.

**Describe the bug**
Although the `isnan` and `isfinite` macros have been specified by ISO C for over 2 decades, there are still some (obsolete) C libraries in use that do not provide these.

**To Reproduce**
Compile LC on VxWorks 6.9, error is reported:
```
apps/lc/fsw/src/lc_watch.c: In function 'LC_FloatCompare':
apps/lc/fsw/src/lc_watch.c:680: error: implicit declaration of function 'isnan'
apps/lc/fsw/src/lc_watch.c: In function 'LC_ValidateWDT':
apps/lc/fsw/src/lc_watch.c:1016: error: implicit declaration of function 'isfinite'
```

**Expected behavior**
Should build successfully

**System observed on:**
VxWorks 6.9, GNU C compiler version 4.3

**Additional context**
This appears to be a bug in the math.h header.  It has these macros defined, but they are conditional on GNUC version <= 2.  As this is using GNU C version 4.3, it does not provide them.  But nothing else does, either.

**Reporter Info**
Joseph Hickey, Vantage Systems, Inc.

--- a/fsw/src/lc_watch.c
+++ b/fsw/src/lc_watch.c
@@ -32,7 +32,26 @@
 #include "lc_perfids.h"
 #include "lc_platform_cfg.h"
 
+#include <float.h>
 #include <math.h>
+
+/*
+ * An ISO-compliant math.h header should provide "isnan" and "isfinite" macros
+ * as they are dicatated by C99.  However some C libraries still in use are not
+ * fully compliant.  If these macros are not defined, define a substitute here.
+ *
+ * Note these are not ideal/complete implementations of these macros, but they
+ * are OK based on the way they are used inside this source file.  Use caution if
+ * changing the code using these, notably:
+ *  - isfinite assumes float type
+ *  - both macros evaluate the argument twice - use only with a simple variable
+ */
+#ifndef isnan
+#define isnan(x) ((x) != (x))
+#endif
+#ifndef isfinite
+#define isfinite(x) ((x) > -FLT_MAX && (x) < FLT_MAX)
+#endif
 
 /* * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
 /*                                                                 */

